isis/tests/SpiceRotationTests.cpp

```diff
@@ -0,0 +1,822 @@
+#include "gmock/gmock.h"
+
+#include <algorithm>
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+#include <vector>
+
+#include <QString>
+
+#include <nlohmann/json.hpp>
+
+#include "Angle.h"
+#include "FileName.h"
+#include "IException.h"
+#include "Preference.h"
+#include "SpiceRotation.h"
+#include "Table.h"
+#include "TestUtilities.h"
+
+// Declarations for bindings for Naif Spicelib routines that do not have
+// a wrapper
+extern int bodeul_(integer *body, doublereal *et, doublereal *ra,
+                   doublereal *dec, doublereal *w, doublereal *lamda);
+
+using json = nlohmann::json;
+using namespace std;
+using namespace Isis;
+
+// Old unit test set output precision to 8 digits.
+double testTolerance = 1e-8;
+
+// Test case is taken from moc red wide angle image ab102401
+// sn = MGS/561812335:32/MOC-WA/RED
+//
+// This is written as a fixture to ensure that the test kernels get unloaded
+// regardless of how the test finishes.
+class SpiceRotationKernels : public ::testing::Test {
+  protected:
+    vector<QString> kernels;
+    double startTime;
+    double endTime;
+    int frameCode;
+    int targetCode;
+
+  void SetUp() {
+    startTime = -69382819.0;
+    endTime = -69382512.0;
+    frameCode = -94031;
+    targetCode = 499;
+
+    QString dir = FileName("$ISISTESTDATA/isis/src/base/unitTestData/kernels").expanded() + "/";
+    kernels.clear();
+    kernels.push_back(dir + "naif0007.tls");
+    kernels.push_back(dir + "MGS_SCLKSCET.00045.tsc");
+    kernels.push_back(dir + "moc13.ti");
+    kernels.push_back(dir + "moc.bc");
+    kernels.push_back(dir + "moc.bsp");
+    kernels.push_back(dir + "de405.bsp");
+    kernels.push_back(FileName("$base/kernels/pck/pck00009.tpc").expanded());
```

Comment:
This kernel is ~100KB, should I add a copy to the unit test data so that all of the kernels used are in there?

Reply:
I don't have a strong opinion about this, but I can see the benefit of having all of the used test kernels in the unit test data directory. Maybe @scsides can chime in?

Reply:
@kaitlyndlee 's makes a good point. It would keep the test running even if 00009 was removed some day.